### PR TITLE
Create a table to store all events #250

### PR DESCRIPTION
**Pull request by linaskitm**
I created a `events` table in 1-schema.sql and added FK `event_id` in these 4 tables  ( `animal_event_general` `animal_event_medical_record`  `animal_event_found` `animal_event_given_away`).   Should I add FK in others tables?

Should I do changes in src/schema/typeDefs/event.graphql by adding `eventsId: Int!`?

### CODE REVIEW

**Comment by linaskitm** [comment]
Screenshot 2021-05-26 140458

Idk why I can't pass Integration Test.

File: database/1-schema.sql [hunk 1]
```diff
@@ -1,5 +1,4 @@
 -- ORGANIZATION
-
```
Comment (AzisK) [inline database/1-schema.sql:2]: I believe these spaces are for better readability, no need to delete them

**Review by AzisK — CHANGES_REQUESTED** [review]
Good job in general.

Did you nuke and restart your local database? I think the initialization script might have failed because other event tables have foreign key in `events` table and that one is empty.

I didn't mention in the ticket but we will get rid of `animal_event_general`, `animal_event_medical_record` etc. event tables later, so no need to add `event_id` in them. We will only have some sort of event_[type]_details tables that will have foreign keys referencing `events` table

Some fake data in `events`, even copying the same events, would be good to have as well

**Comment by linaskitm** [comment]
Thanks for review.
It looks like I wasn't restarted my local database in first place.
Added some fake data and did restore spaces for better readability.

File: database/1-schema.sql [hunk 2]
```diff
@@ -335,48 +334,63 @@ CREATE TABLE animal_event_given_away (
...
+    comments TEXT
+);
+
 -- DATE UPDATES
 
 CREATE EXTENSION moddatetime;
 
-CREATE TRIGGER organization_mod_time BEFORE UPDATE ON organization
-FOR EACH ROW EXECUTE PROCEDURE moddatetime (mod_time);
+CREATE TRIGGER organization_mod_time BEFORE
```
Comment (AzisK) [inline database/1-schema.sql:353]: Why change?

Comment (AzisK) [reply database/1-schema.sql:353]: It's hard to follow changes this way

File: database/3-fake-data.sql [hunk 3]
```diff
@@ -92,8 +99,7 @@ INSERT INTO animal_event_general
 (5, 'LocationChange', 1.00, '2018-07-31', 'Pasikeitė laikymo vietos adresas.', '0932hfdsa'),
 (5, 'LocationChange', 1.00, '2018-08-01', 'Pasikeitė laikymo vietos adresas.', 'afhu9w4f78'),
 (5, 'LocationChange', 1.00, '2018-08-02', 'Pasikeitė laikymo vietos adresas.', 'dhjbwau74a6'),
-(5, 'LocationChange', 1.00, '2018-08-03', 'Pasikeitė laikymo vietos adresas.', 'dhjbwau74a6'),
-(5, 'LocationChange', 1.00, '2018-08-04', 'Pasikeitė laikymo vietos adresas.', 'afhu9w4f78');
+(5, 'LocationChange', 1.00, '2018-08-03', 'Pasikeitė laikymo vietos adresas.', 'dhjbwau74a6'),(5, 'LocationChange', 1.00, '2018-08-04', 'Pasikeitė laikymo vietos adresas.', 'afhu9w4f78');
```
Comment (AzisK) [inline database/3-fake-data.sql:102]: I believe there is no need to change this as well

File: database/3-fake-data.sql [hunk 4]
```diff
@@ -60,6 +60,13 @@ INSERT INTO animal_microchip
 (4, '29387', '5', '2020-04-14', '3', DEFAULT),
 (3, '2893402', '6', '2020-03-01', '4', DEFAULT);
 
+INSERT INTO events 
+(animal_id, group_enum, type, date_time, author, comments) VALUES
+(1, 'Medical', 'Prophylaxis', '2020-05-21', '278y2378ryb', 'Yearly vaccine'),
+(2, 'Medical', 'Surgery', '2020-05-22', 'aiubfaw4io09', 'Fast recovery after surgey'),
+(3, 'Medical', 'Medication', '2020-05-21', 'dhjbwau74a6', 'Tick medications'),
+(4, 'Medical', 'Prophylaxis', '2020-05-21', '278y2378ryb', 'Yearly vaccine');
```
Comment (AzisK) [inline database/3-fake-data.sql:68]: Good job

File: database/1-schema.sql (hunk 2 above)
Comment (linaskitm) [reply database/1-schema.sql]: My bad.
I didn't change on purpose, I think my editor changed that after save.

File: database/1-schema.sql [hunk 5]
```diff
@@ -345,38 +356,41 @@ FOR EACH ROW EXECUTE PROCEDURE moddatetime (mod_time);
 CREATE TRIGGER app_user_mod_time BEFORE UPDATE ON app_user
 FOR EACH ROW EXECUTE PROCEDURE moddatetime (mod_time);
 
-CREATE TRIGGER animal_mod_time BEFORE UPDATE ON animal
+CREATE TRIGGER animal_mod_time BEFORE UPDATE ON animal
```
Comment (AzisK) [inline database/1-schema.sql:359]: These spaces at the end just create unnecessary clutter in git but otherwise things look good

**Review by AzisK — APPROVED** [review]

File: database/1-schema.sql [hunk 6]
```diff
@@ -335,6 +334,18 @@ CREATE TABLE animal_event_given_away (
     comments TEXT
 );
 
+CREATE TABLE events (
+    id SERIAL PRIMARY KEY,
+    animal_id INTEGER REFERENCES animal(id) ON DELETE CASCADE NOT NULL,
+    group_enum event_group,
```
Comment (ignasposka) [inline database/1-schema.sql:340]: Why we cannot call it just `group` instead of `group_enum`?

**Review by ignasposka — CHANGES_REQUESTED** [review]

File: database/1-schema.sql (hunk 6 above)
Comment (linaskitm) [reply database/1-schema.sql:340]: I tried call `group` but after running docker I got this syntax error:

Screenshot 2021-05-31 151208

I changed into  `group_enum` and worked.

**Comment by linaskitm** [comment]
Should I or reviewer resolve the conflicts in `database/1-schema.sql`?

**Comment by AzisK** [comment]
> Should I or reviewer resolve the conflicts in `database/1-schema.sql`?

Yes. Somebody has to if we want to merge this pull request

**Comment by linaskitm** [comment]
> > Should I or reviewer resolve the conflicts in `database/1-schema.sql`?
> 
> Yes. Somebody has to if we want to merge this pull request

**Review by ignasposka — APPROVED** [review]
